wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/model/impl/WorkspaceRuntimeImpl.java

```diff
@@ -47,9 +47,11 @@ public WorkspaceRuntimeImpl(String activeEnv,
         if (devMachine != null) {
             this.devMachine = new MachineImpl(devMachine);
         }
-        this.machines = machines.stream()
-                                .map(MachineImpl::new)
-                                .collect(toList());
+        if (machines != null) {
```

Comment:
It is possible to pass the null to the POJO constructor or setter where collection is required, BUT the getter of the collection must return non-null collection anyway [example](https://github.com/eclipse/che/blob/21d59d72cd8f6c7636f7f679d8208ca8fa913475/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/model/impl/EnvironmentImpl.java#L42-L46) that's how we avoid npes is such usecases.
This change is a good one and should come with a getter changes as well.